docs/articles/documentation/guides/basic-guides/select-page-elements.md

```diff
@@ -0,0 +1,542 @@
+
+An error can also occur when you call [selector's methods](#member-tables) in a chain. These methods are applied to the selector one by one. TestCafe detects a method after which the selector no longer matches any DOM element. This method is highlighted in the error message.
+
+![Selector methods in a report](../../../../images/failed-selector-report.png)
+
+## Extend Selectors with Custom Properties and Methods
+
+You can add custom properties and methods to TestCafe selectors. Custom members allow you to implement framework-specific API and retrieve DOM element properties not included in standard selectors.
+
+To add a custom member, provide client-side code for the method or property getter with the [addCustomMethods](../../reference/test-api/selector/addcustommethods.md) or [addCustomDOMProperties](../../reference/test-api/selector/addcustomdomproperties.md) method.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    let fieldSet = Selector('fieldset').addCustomMethods({
+        getLabel: (el, idx) => {
+            return el[0].elements[idx].labels[0];
+        }
+    }, {
+        returnDOMNodes: true
+    });
+
+    await t.expect(fieldSet.nth(1).getLabel(3).textContent).eql('Easy embedding into a Continuous integration system');
+
+    fieldSet = fieldSet.addCustomDOMProperties({
+        legend: el => el.querySelector('legend').innerText
+    });
+
+    await t.expect(fieldSet.nth(1).legend).eql('Which features are important to you:');
+});
+```
+
+Note that custom properties and methods propagate through the selector chain. For instance, you can define `selector.myProperty` and access it further in the chain: `selector.nth(2).myProperty`, `selector.withText('ABC').myProperty`, etc.
+
+## Overwrite Options
+
+You can use the selector's [with](../../reference/test-api/selector/with.md) method to overwrite its [options](../../reference/test-api/selector/with.md#options). This allows you to derive selectors with different settings from the same base selector.
+
+```js
+import { Selector } from 'testcafe';
+
+const elementWithId = Selector(id => document.getElementById(id));
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const visibleElementWithId = elementWithId.with({
+        visibilityCheck: true
+    });
+
+    const visibleButton = await visibleElementWithId('submit-button');
+});
+```
+
+## Framework-Specific Selectors
+
+TestCafe selector functions are based on CSS selectors or client JS code, which suit for any HTML5 website. However, if you use a front-end framework, you may want your tests to leverage the framework architecture specifics to locate elements more precisely. For instance, you could use the component tree for React or element bindings for Aurelia.
```

Comment:
which suit for?

Reply:
to leverage?

Reply:
TestCafe selector functions are based on CSS selectors or client JS code. This approach fits any HTML5 website. However, if you use a front-end framework, you may want your selectors to be based on the framework-specific entities to locate elements more precisely.